etcd fileutil unit test failing

<!-- This form is for bug reports and feature requests ONLY! 

If you're looking for help check [Stack Overflow](https://stackoverflow.com/questions/tagged/kubernetes) and the [troubleshooting guide](https://kubernetes.io/docs/tasks/debug-application-cluster/troubleshooting/).
-->

**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug
/sig api-machinery

**What happened**:
pull-kubernetes-bazel job is failing
See https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/49507/pull-kubernetes-bazel/32757/

```
--- FAIL: TestIsDirWriteable (0.00s)
	fileutil_test.go:38: expected IsDirWriteable to error
2017-07-29 00:32:54.490716 I | fileutil: purged file /tmp/purgefile088685595/0.test successfully
2017-07-29 00:32:54.490758 I | fileutil: purged file /tmp/purgefile088685595/1.test successfully
2017-07-29 00:32:54.490777 I | fileutil: purged file /tmp/purgefile088685595/2.test successfully
2017-07-29 00:32:54.501104 I | fileutil: purged file /tmp/purgefile088685595/3.test successfully
2017-07-29 00:32:54.511479 I | fileutil: purged file /tmp/purgefile088685595/4.test successfully
2017-07-29 00:32:54.521806 I | fileutil: purged file /tmp/purgefile088685595/5.test successfully
2017-07-29 00:32:54.532322 I | fileutil: purged file /tmp/purgefile088685595/6.test successfully
2017-07-29 00:32:54.542918 I | fileutil: purged file /tmp/purgefile586492606/0.test successfully
2017-07-29 00:32:54.542962 I | fileutil: purged file /tmp/purgefile586492606/1.test successfully
2017-07-29 00:32:54.542992 I | fileutil: purged file /tmp/purgefile586492606/2.test successfully
2017-07-29 00:32:54.543034 I | fileutil: purged file /tmp/purgefile586492606/3.test successfully
2017-07-29 00:32:54.543076 I | fileutil: purged file /tmp/purgefile586492606/4.test successfully
2017-07-29 00:32:54.565204 I | fileutil: purged file /tmp/purgefile586492606/5.test successfully
2017-07-29 00:32:54.565253 I | fileutil: purged file /tmp/purgefile586492606/6.test successfully
FAIL
```

**What you expected to happen**:
test should pass

**How to reproduce it (as minimally and precisely as possible)**:
bazel test //third_party/forked/etcd221/pkg/fileutil:go_default_test

--- a/build/root/.bazelrc
+++ b/build/root/.bazelrc
@@ -7,3 +7,7 @@ build --workspace_status_command hack/print-workspace-status.sh
 
 # Make /tmp hermetic
 build --sandbox_tmpfs_path=/tmp
+
+# Ensure that Bazel never runs as root, which can cause unit tests to fail.
+# This flag requires Bazel 0.5.0+
+build --sandbox_fake_username

